subsys/bluetooth/controller/hal/nrf5/radio/radio.c

```diff
@@ -792,21 +818,41 @@ void radio_tmr_end_capture(void)
 	HAL_RADIO_END_TIME_CAPTURE_PPI_REGISTER_TASK =
 		HAL_RADIO_END_TIME_CAPTURE_PPI_TASK;
 	NRF_PPI->CHENSET = HAL_RADIO_END_TIME_CAPTURE_PPI_ENABLE;
+
+#if  defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+	HAL_EVENT_TIMER_CLEAR_PPI_REGISTER_EVT = HAL_EVENT_TIMER_CLEAR_PPI_EVT;
+	HAL_EVENT_TIMER_CLEAR_PPI_REGISTER_TASK =
+		HAL_EVENT_TIMER_CLEAR_PPI_TASK;
+	NRF_PPI->CHENSET = HAL_EVENT_TIMER_CLEAR_PPI_ENABLE;
+#endif /* defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) */
+
 }
 
 u32_t radio_tmr_end_get(void)
 {
+#if defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+	return last_pdu_end_us;
+#else /* defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) */
 	return EVENT_TIMER->CC[2];
+#endif /* defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) */
 }
 
+static u32_t tmr_sample_val;
```

Comment:
we tend to define globals at the top of the file

Reply:
agree with the general concept, of course. However:
1. this variable is only used in these two functions.
2. if we move to top, we need _yet_ another #ifdef CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER
3. there is already another static variable, sw_tifs_toggle that is defined right before it is used.
So, @carlescufi I would recommend to reduce noise, and keep it here.

Reply:
Ok, makes sense.